deepchem/models/torch_models/layers.py

```diff
@@ -5000,21 +5008,17 @@ def forward(self, one_electron: torch.Tensor, two_electron: torch.Tensor):
                     two_electron[:, i, self.spin[0]:, :], dim=1)
                 f: torch.Tensor = torch.cat((one_electron[:, i, :], g_one_up,
                                              g_one_down, g_two_up, g_two_down),
-                                            dim=1)
+                                            dim=1).double()
                 if l == 0 or (self.n_one[l] != self.n_one[l - 1]) or (
                         self.n_two[l] != self.n_two[l - 1]):
-                    one_electron_tmp[:, i, :] = torch.tanh(self.v[l](f.to(
-                        torch.float32)))
+                    one_electron_tmp[:, i, :] = torch.tanh(self.v[l](f))
```

Comment:
Can you explain the changes in this section?

Reply:
Here the tensor was already in float32, so ".to(torch.float32)" adds redundancy